swap_meet/vendor.py

```diff
@@ -1,2 +1,121 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = [] if inventory is None else inventory
+        
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_id(self, item_id):
+        for item in self.inventory:
+            if item.id == item_id:
+                return item
+        return None
+    
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        """
+        If this Vendor's inventory contain my_item and 
+        the friend's inventory contain their_item. 
+        The method returns True,
+        else the method returns False.
+        """
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.inventory.remove(my_item) 
+            other_vendor.inventory.append(my_item)
+            # if their_item in other_vendor.inventory:
+            other_vendor.remove(their_item) 
+            self.inventory.append(their_item)
+            return True
+        else: 
+            return False
+            
+    def swap_first_item(self, other_vendor):
+        """
+        Function swaps my first item with their first item.
+        
+        Input: other vendor
+
+        Output: Returns either True or False. 
+        - Returns False if my inventory or their inventory is empty. 
+        - Otherwise, returns True.        
+        """
+
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        
+        first_item_in_mine = self.inventory[0]
+        first_item_in_vendor = other_vendor.inventory[0]
+        
+        self.swap_items(other_vendor, first_item_in_mine, first_item_in_vendor)
+        
+        return True
+
+
+    def get_by_category(self, category):
+        """
+        Returns a list of items aligned to the same category. 
+
+        Input: a string, representing a category.
+
+        Output: returns a inventory list of items with the same category.
+        """
+
+        item_list_by_category = []
+
+        for item in self.inventory:
+            if item.get_category() == category:
+                item_list_by_category.append(item)
+
+        return item_list_by_category
+
+    
+    def get_best_by_category(self, category):
+        """
+        Finds item with the best condition in a certain category 
+
+        Input: a string, representing a category
+
+        Output: returns single item with the highest condition and same category
+        """
+
+        items = self.get_by_category(category)
+
+        if not items:
+            return None
+        
+        best_item = items[0]
+        for item in items:
+            if item.condition > best_item.condition:
```

Comment:
I love what y'all did when it comes to checking the condition. You recognized that the best_item will hold the best_condition data so you don't need to hold it separately! Well done!